addons/web/static/src/core/domain.js

```diff
@@ -199,10 +199,10 @@ function matchCondition(record, condition) {
     switch (operator) {
         case "=":
         case "==":
-            return fieldValue === value;
+            return JSON.stringify(fieldValue) === JSON.stringify(value);
```

Comment:
doesn't look like the proper fix in my opinion. This could crash if the value is not stringifiable (if it contains some object with a cycle). I think that we should instead do the comparison in the case of array by hand (checking the length, then iterating on each value shallowly (?))